sorting/multi_threaded/merge_sort/multi_threaded_merge_sort.go

```diff
@@ -0,0 +1,146 @@
+package main
+
+import (
+	"fmt"
+	"math/rand"
+	"time"
+)
+
+const threshold = 10000
+
+func parallelMergeSort(arr []int, c chan []int) {
+	if len(arr) <= 1 {
+		c <- arr
+		return
+	}
+
+	if len(arr) <= threshold {
+		// mergeSort(arr)
+		timSort(arr)
+		c <- arr
+		return
+	}
+
+	middleIndex := len(arr) / 2
+	leftChan := make(chan []int)
+	rightChan := make(chan []int)
+
+	go parallelMergeSort(arr[:middleIndex], leftChan)
+	go parallelMergeSort(arr[middleIndex:], rightChan)
+
+	leftSorted := <-leftChan
+	rightSorted := <-rightChan
+
+	c <- merge(leftSorted, rightSorted)
+}
+
+func merge(leftArray, rightArray []int) []int {
+	result := make([]int, len(leftArray)+len(rightArray))
+	leftIndex, rightIndex := 0, 0
+	for i := 0; i < len(result); i++ {
+		if leftIndex >= len(leftArray) {
+			result[i] = rightArray[rightIndex]
+			rightIndex++
+			continue
+		}
+		if rightIndex >= len(rightArray) {
+			result[i] = leftArray[leftIndex]
+			leftIndex++
+			continue
+		}
+		if leftArray[leftIndex] < rightArray[rightIndex] {
+			result[i] = leftArray[leftIndex]
+			leftIndex++
+		} else {
+			result[i] = rightArray[rightIndex]
+			rightIndex++
+		}
+	}
+	return result
+}
+
+func mergeSort(inputArray []int) []int {
+	if len(inputArray) <= 1 {
+		return inputArray
+	}
+
+	middleIndex := len(inputArray) / 2
+	leftArray := mergeSort(inputArray[:middleIndex])
+	rightArray := mergeSort(inputArray[middleIndex:])
+
+	return merge(leftArray, rightArray)
+}
```

Comment:
remove this.

Reply:
But I am using `mergesort` to compare with multi-threaded mergesort. 
So instead I'll use `tim sort` to compare with multi-threaded mergesort ?

Reply:
Yes